sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -21,6 +22,85 @@
     from azure.core.pipeline import PipelineRequest, PipelineResponse
 
 
+class _HttpChallenge(object):
+    """ Represents a parsed HTTP WWW-Authentication Bearer challenge from a server. """
+
+    def __init__(self, request_uri, challenge):
+        self.source_authority = self._get_request_authority(request_uri)
+        self.source_uri = request_uri
+        self._parameters = {}
+
+        # Get the scheme of the challenge and remove it from the challenge string
+        trimmed_challenge = self._get_challenge_parts(challenge)
+        split_challenge = trimmed_challenge.split(" ", 1)
+        self.scheme = split_challenge[0]
+        trimmed_challenge = split_challenge[1]
+
+        # Split trimmed challenge into name=value pairs; these pairs are expected to be split by either commas or spaces
+        # Values may be surrounded by quotes, which are stripped here
+        separator = "," if "," in trimmed_challenge else " "
+        for item in trimmed_challenge.split(separator):
+            # process name=value pairs
+            comps = item.split("=")
+            if len(comps) == 2:
+                key = comps[0].strip(' "')
+                value = comps[1].strip(' "')
+                if key:
+                    self._parameters[key] = value
+
+        if not self._parameters:
+            raise ValueError("Invalid challenge parameters")
+
+        # Challenge must specify authorization or authorization_uri
+        if "authorization" not in self._parameters and "authorization_uri" not in self._parameters:
+            raise ValueError("Invalid challenge parameters")
+
+        authorization_uri = self.get_authorization_server()
+        # the authorization server URI should look something like https://login.windows.net/tenant-id[/oauth2/authorize]
+        uri_path = urllib.parse.urlparse(authorization_uri).path.lstrip("/")
+        self.tenant_id = uri_path.split("/")[0] or None
+
+    def get_authorization_server(self):
```

Comment:
Is this method supposed to be internal-public?

Reply:
This (and other methods on the class) came directly from Key Vault's implementation of a [similar class](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/http_challenge.py). This doesn't need to be public -- I can make this private

Reply:
On second thought, I'm removing this function entirely. Doing this on line 58 is easier to read and more efficient